tests/unittests/algo/test_tpe.py

```diff
@@ -0,0 +1,220 @@
+#!/usr/bin/env python
+# -*- coding: utf-8 -*-
+"""Tests for :mod:`orion.algo.tpe`."""
+
+import numpy
+import pytest
+
+from orion.algo.space import Integer, Real, Space
+from orion.algo.tpe import adaptive_parzen_estimator, compute_max_ei_point, GMMSampler, TPE
+
+
+@pytest.fixture()
+def space():
+    """Return an optimization space"""
+    space = Space()
+    dim1 = Real('yolo1', 'uniform', -3, 6)
+    space.register(dim1)
+    dim2 = Real('yolo2', 'uniform', -2, 4)
+    space.register(dim2)
+
+    return space
+
+
+@pytest.fixture
+def tpe(space):
+    """Return an instance of TPE."""
+    return TPE(space)
+
+
+@pytest.fixture
+def gmm_sampler(tpe):
+    """Return an instance of GMMSampler"""
+    mus = numpy.linspace(-3, 3, num=12, endpoint=False)
+    sigmas = [0.5] * 12
+
+    return GMMSampler(tpe, mus, sigmas, -3, 3)
+
+
+def test_compute_max_ei_point():
+    """Test that max ei point is computed correctly"""
+    points = numpy.linspace(-3, 3, num=10)
+    below_likelis = numpy.linspace(0.5, 0.9, num=10)
+    above_likes = numpy.linspace(0.2, 0.5, num=10)
+    max_ei_point = compute_max_ei_point(points, below_likelis, above_likes)
+    assert max_ei_point == 3
+
+
+def test_adaptive_parzen_normal_estimator():
+    """Test adaptive kernel estimator"""
+    obs_mus = numpy.linspace(-1, 5, num=30, endpoint=False)
+    low = -1
+    high = 5
+    mus, sigmas, weights = adaptive_parzen_estimator(obs_mus, low, high, prior_weight=1.0,
+                                                     equal_weight=False, flat_num=25)
+
+    assert len(mus) == len(sigmas) == len(weights) == 30 + 1
+    assert numpy.all(weights[-25:] == weights[-1])
+    assert numpy.all(sigmas == 6 / 10)
+
+    # equal weight
+    mus, sigmas, weights = adaptive_parzen_estimator(obs_mus, low, high, prior_weight=1.0,
+                                                     equal_weight=True, flat_num=25)
+    assert numpy.all(weights[-25:] == weights[-1])
+    assert numpy.all(sigmas == 6 / 10)
+
+    # priori weight
+    mus, sigmas, weights = adaptive_parzen_estimator(obs_mus, low, high, prior_weight=0.5,
+                                                     equal_weight=False, flat_num=25)
+    priori_pos = numpy.searchsorted(mus, 2)
+    assert weights[priori_pos] == 0.5 / 31
+    assert numpy.all(weights[31 - priori_pos:] == weights[-1])
+    assert numpy.all(sigmas == 6 / 10)
+
+    # sigma clip
```

Comment:
Sigmas are a bit tricky with edges being different than middle ones. I would test them separately in another unit-test and verify the edges plus the middle ones. That would require (pseudo)-random intervals as well, not `linear()`.